internal/controllers/machine/machine_controller_phases_test.go

```diff
@@ -769,55 +747,40 @@ func TestReconcileBootstrap(t *testing.T) {
 					"dataSecretName": "secret-data",
 				},
 			},
-			machine: &clusterv1.Machine{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "bootstrap-test-existing",
-					Namespace: metav1.NamespaceDefault,
-				},
-				Spec: clusterv1.MachineSpec{
-					Bootstrap: clusterv1.Bootstrap{
-						ConfigRef: &corev1.ObjectReference{
-							APIVersion: "bootstrap.cluster.x-k8s.io/v1beta1",
-							Kind:       "GenericBootstrapConfig",
-							Name:       "bootstrap-config1",
-						},
-						DataSecretName: ptr.To("secret-data"),
-					},
-				},
-				Status: clusterv1.MachineStatus{
-					BootstrapReady: true,
-				},
-			},
-			expectResult: ctrl.Result{},
-			expectError:  false,
+			bootstrapConfigGetError: nil,
+			expectResult:            ctrl.Result{},
+			expectError:             false,
 			expected: func(g *WithT, m *clusterv1.Machine) {
 				g.Expect(m.Status.BootstrapReady).To(BeTrue())
-				g.Expect(*m.Spec.Bootstrap.DataSecretName).To(BeEquivalentTo("secret-data"))
+				g.Expect(m.Spec.Bootstrap.DataSecretName).NotTo(BeNil())
+				g.Expect(*m.Spec.Bootstrap.DataSecretName).To(ContainSubstring("secret-data"))
 			},
 		},
 		{
-			name: "existing machine, bootstrap provider is not ready, and ownerref updated",
+			name:    "bootstrap config ready with no bootstrap secret",
```

Comment:
Today a bootstrap secret can report ready but without a bootstrap data secret name, I don't think this should be accepted

Reply:
Agreed

Reply:
my bad, we are already erroring in this case